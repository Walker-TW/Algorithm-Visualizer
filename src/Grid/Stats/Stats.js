import React, { useState } from 'react';
import './Stats.css';

const Stats = () => {
  const [collapsed, setCollapsed] = useState(true);

  return collapsed ? (
    <div
      onClick={() => {
        setCollapsed(false);
      }}
      className={'alt-collapsed'}
      children={'stats'}
    />
  ) : (
    <div className={'stats'}>
      <div
        onClick={() => {
          setCollapsed(true);
        }}
        style={{
          width: '1rem',
          height: '1.4rem',
          backgroundColor: 'red',
          float: 'right',
          textAlign: 'center',
          borderRadius: '4px',
        }}
        children={'X'}
      />
      <p>How to use:</p>
      <ul>
        <li>First click sets the start node</li>
        <li>Second sets the end node</li>
        <li>Click Run Dijkstra to run the algorithm!</li>
      </ul>
    </div>
  );
};

export default Stats;
