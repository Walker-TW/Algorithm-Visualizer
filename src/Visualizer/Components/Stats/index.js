import React from 'react';
import { Table } from 'react-bootstrap';
import './Stats.css';

const Stats = ({ runtime, nodesProccessed, fastestPath, algorithmRan }) => {
  return (
    <Table striped bordered hover variant="dark" size="sm">
      <tbody>
        <tr>
          <th>Algorithm</th>
          <th>Runtime (Microseconds)</th>
          <th>Nodes Processed</th>
          <th>Fastest Path Total</th>
        </tr>
        <tr>
          <td>{algorithmRan}</td>
          <td>{runtime}</td>
          <td>{nodesProccessed}</td>
          <td>{fastestPath}</td>
        </tr>
      </tbody>
    </Table>
  );
};

export { Stats };
