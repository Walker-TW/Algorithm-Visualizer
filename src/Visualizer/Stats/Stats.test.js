import React from "react";
import Stats from "./Stats";
import { shallowToJson } from "enzyme-to-json";
import { shallow, mount, render } from "enzyme";

describe("<Stats />", () => {
  it("renders", () => {
    const wrapper = shallow(<Stats />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
