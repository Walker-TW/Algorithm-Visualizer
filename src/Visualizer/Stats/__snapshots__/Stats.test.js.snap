// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stats /> renders 1`] = `
<ForwardRef
  bordered={true}
  hover={true}
  size="sm"
  striped={true}
  variant="dark"
>
  <tbody>
    <tr>
      <th>
        Algorithm
      </th>
      <th>
        Runtime (Microseconds)
      </th>
      <th>
        Nodes Processed
      </th>
      <th>
        Fastest Path Total
      </th>
    </tr>
    <tr>
      <td />
      <td />
      <td />
      <td />
    </tr>
  </tbody>
</ForwardRef>
`;
